Sources/GRPC/GRPCClientStateMachine.swift

```diff
@@ -0,0 +1,588 @@
+  ) -> Result<HTTPHeaders, RecieveResponseHeadError> {
+    let result: Result<HTTPHeaders, RecieveResponseHeadError>
+
+    switch self {
+    case let .clientStreamingServerIdle(writeState, responseArity):
+      switch self.parseResponseHeaders(responseHead, responseArity: responseArity, logger: logger) {
+      case .success(let readState):
+        self = .clientStreamingServerStreaming(writeState, readState)
+        result = .success(responseHead.headers)
+      case .failure(let error):
+        result = .failure(error)
+      }
+
+    case let .clientClosedServerIdle(responseArity):
+      switch self.parseResponseHeaders(responseHead, responseArity: responseArity, logger: logger) {
+      case .success(let readState):
+        self = .clientClosedServerStreaming(readState)
+        result = .success(responseHead.headers)
+      case .failure(let error):
+        result = .failure(error)
+      }
+
+    case .clientIdleServerIdle,
+         .clientClosedServerStreaming,
+         .clientStreamingServerStreaming,
+         .clientClosedServerClosed:
+      result = .failure(.invalidState)
+    }
+
+    return result
+  }
+
+  /// See `GRPCClientStateMachine.recieveResponse(_:)`.
+  mutating func recieveResponse(
+    _ buffer: inout ByteBuffer
+  ) -> Result<[Response], MessageReadError> {
+    let result: Result<[Response], MessageReadError>
+
+    switch self {
+    case .clientClosedServerStreaming(var readState):
+      result = readState.readMessage(&buffer, as: Response.self)
+      self = .clientClosedServerStreaming(readState)
+
+    case .clientStreamingServerStreaming(let writeState, var readState):
+      result = readState.readMessage(&buffer, as: Response.self)
+      self = .clientStreamingServerStreaming(writeState, readState)
+
+    case .clientIdleServerIdle,
+         .clientStreamingServerIdle,
+         .clientClosedServerIdle,
+         .clientClosedServerClosed:
+      result = .failure(.invalidState)
+    }
+
+    return result
+  }
+
+  /// See `GRPCClientStateMachine.recieveEndOfResponseStream(_:)`.
+  mutating func recieveEndOfResponseStream(
+    _ trailers: HTTPHeaders
+  ) -> Result<GRPCStatus, InvalidStateError> {
+     let result: Result<GRPCStatus, InvalidStateError>
+
+     switch self {
+     case .clientStreamingServerStreaming,
+          .clientStreamingServerIdle,
+          .clientClosedServerIdle,
+          .clientClosedServerStreaming:
+       result = .success(self.parseTrailers(trailers))
+       self = .clientClosedServerClosed
+
+     case .clientIdleServerIdle,
+          .clientClosedServerClosed:
+      result = .failure(.invalidState())
+     }
+
+     return result
+   }
+
+  /// Makes the request headers ("Request-Headers" in the specification) used to initiate an RPC
+  /// call.
+  ///
+  /// See: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests
+  ///
+  /// - Important: pseudo-headers (":method", ":scheme", ":path") are not included here and should
```

Comment:
super-nit (optional): use backticks instead of quotes?

Reply:
Was this removed altogether?

Reply:
Yes: they're added as part of the `HTTPRequestHead` and made into pseudo-headers by the H2 to H1 codec so I didn't think this was necessary.